Content.Client/UserInterface/CharacterSetupGui.cs

```diff
@@ -0,0 +1,268 @@
+            });
+
+            _createNewCharacterButton = new Button
+            {
+                Text = "Create new slot...",
+                ToolTip = $"A maximum of {preferencesManager.Settings.MaxCharacterSlots} characters are allowed."
+            };
+            _createNewCharacterButton.OnPressed += args =>
+            {
+                preferencesManager.CreateCharacter(HumanoidCharacterProfile.Default());
+                UpdateUI();
+            };
+
+            hBox.AddChild(new PanelContainer
+            {
+                PanelOverride = new StyleBoxFlat {BackgroundColor = NanoStyle.NanoGold},
+                CustomMinimumSize = (2, 0)
+            });
+            _humanoidProfileEditor = new HumanoidProfileEditor(localization, preferencesManager);
+            _humanoidProfileEditor.OnProfileChanged += newProfile => { UpdateUI(); };
+            hBox.AddChild(_humanoidProfileEditor);
+
+            UpdateUI();
+        }
+
+        private void UpdateUI()
+        {
+            var numberOfFullSlots = 0;
+            var characterButtonsGroup = new ButtonGroup();
+            _charactersVBox.RemoveAllChildren();
+            var characterIndex = 0;
+            foreach (var character in _preferencesManager.Preferences.Characters)
+            {
+                if (character is null)
+                {
+                    characterIndex++;
+                    continue;
+                }
+
+                numberOfFullSlots++;
+                var characterPickerButton = new CharacterPickerButton(_entityManager,
+                    _preferencesManager,
+                    characterButtonsGroup,
+                    character);
+                _charactersVBox.AddChild(characterPickerButton);
+
+                var characterIndexCopy = characterIndex;
+                characterPickerButton.ActualButton.OnPressed += args =>
+                {
+                    _humanoidProfileEditor.Profile = (HumanoidCharacterProfile) character;
+                    _humanoidProfileEditor.CharacterSlot = characterIndexCopy;
+                    _humanoidProfileEditor.UpdateControls();
+                    _preferencesManager.SelectCharacter(character);
+                };
+                characterIndex++;
+            }
+
+            _createNewCharacterButton.Disabled =
+                numberOfFullSlots >= _preferencesManager.Settings.MaxCharacterSlots;
+            _charactersVBox.AddChild(_createNewCharacterButton);
+        }
+
+        private class CharacterPickerButton : Control
+        {
+            public readonly Button ActualButton;
+            private IEntity _previewDummy;
+
+            public CharacterPickerButton(
+                IEntityManager entityManager,
+                IClientPreferencesManager preferencesManager,
+                ButtonGroup group,
+                ICharacterProfile profile)
+            {
+                _previewDummy = entityManager.SpawnEntityAt("HumanMob_Content",
```

Comment:
Would it make sense to spawn a type of dummy that only has the components necessary to generate the preview? So no stuff like hunger component and whatever.

Reply:
Yes, definitely